src/vgg.jl

```diff
@@ -0,0 +1,156 @@
+    Conv(ws[:conv3_3_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv3_3_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    Conv(ws[:conv3_4_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv3_4_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    x -> maxpool(x, (2,2)),
+    Conv(ws[:conv4_1_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv4_1_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    Conv(ws[:conv4_2_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv4_2_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    Conv(ws[:conv4_3_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv4_3_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    Conv(ws[:conv4_4_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv4_4_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    x -> maxpool(x, (2,2)),
+    Conv(ws[:conv5_1_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv5_1_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    Conv(ws[:conv5_2_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv5_2_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    Conv(ws[:conv5_3_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv5_3_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    Conv(ws[:conv5_4_w_0][end:-1:1,:,:,:][:,end:-1:1,:,:], ws[:conv5_4_b_0], relu, pad = (1,1), stride = (1,1), dilation = (1,1)),
+    x -> maxpool(x, (2,2)),
+    x -> reshape(x, :, size(x, 4)),
+    Dense(ws[:fc6_w_0]', ws[:fc6_b_0], relu),
+    Dropout(0.5f0),
+    Dense(ws[:fc7_w_0]', ws[:fc7_b_0], relu),
+    Dropout(0.5f0),
+    Dense(ws[:fc8_w_0]', ws[:fc8_b_0]),
+    softmax)
+  Flux.testmode!(ls)
+  return ls
+end
+
+function load_vgg(arr, batchnorm::Bool = false)
+  layers = []
+  in_chs = 3
+  for i in arr
+    if i != 0
+      push!(layers, Conv((3, 3), in_chs=>i, pad = (1, 1)))
+      if batchnorm
+        push!(layers, BatchNorm(i))
+      end
+      push!(layers, x -> relu.(x))
+      in_chs = i
+    else
+      push!(layers, x -> maxpool(x, (2, 2)))
+    end
+  end
+  push!(layers, [x -> reshape(x, :, size(x, 4)), Dense(25088, 4096, relu), Dropout(0.5),
+                 Dense(4096, 4096, relu), Dropout(0.5), Dense(4096, 1000), softmax]...)
+  Chain(layers...)
+end
+
+vgg_configs =
+  Dict("vgg11" => [64, 0, 128, 0, 256, 256, 0, 512, 512, 0, 512, 512, 0],
+       "vgg13" => [64, 64, 0, 128, 128, 0, 256, 256, 0, 512, 512, 0, 512, 512, 0],
+       "vgg16" => [64, 64, 0, 128, 128, 0, 256, 256, 256, 0, 512, 512, 512, 0, 512, 512, 512, 0],
+       "vgg19" => [64, 64, 0, 128, 128, 0, 256, 256, 256, 256, 0, 512, 512, 512, 512, 0, 512, 512, 512, 512, 0])
+
+struct VGG11 <: ClassificationModel{ImageNet.ImageNet1k}
+  layers::Chain
+end
+
+VGG11() = VGG11(load_vgg(vgg_configs["vgg11"]))
+
+Base.show(io::IO, ::VGG11) = print(io, "VGG11()")
+
+Flux.treelike(VGG11)
+
+(m::VGG11)(x) = m.layers(x)
+
+struct VGG11_BN <: ClassificationModel{ImageNet.ImageNet1k}
```

Comment:
Rather than having `_BN`, it'd be nicer to call `VGG19(batchnorm = true/false)`.